ROOTConfig.cmake fails due to mismatching nlohmann_json

<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->

- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

Attempting to build a project that depends on ROOT and uses cmake results in the following error:

```
CMake Error at /home/duncan/opt/mambaforge/conda-bld/omicron_1642421303080/_build_env/share/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  Could not find a configuration file for package "nlohmann_json" that is
  compatible with requested version "3.10.5".

  The following configuration files were considered but not accepted:

    /home/duncan/opt/mambaforge/conda-bld/omicron_1642421303080/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake, version: 3.10.3

Call Stack (most recent call first):
  /home/duncan/opt/mambaforge/conda-bld/omicron_1642421303080/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/cmake/ROOTConfig.cmake:114 (find_dependency)
  src/CMakeLists.txt:13 (find_package)


```

Does this recipe need to pin explicitly to the relevant compatible versions of nlohmann_json?

<br/>
Environment (<code>conda list</code>):
<details>

From mambabuild output

```

    _libgcc_mutex:                        0.1-conda_forge            conda-forge
    _openmp_mutex:                        4.5-1_gnu                  conda-forge
    _sysroot_linux-64_curr_repodata_hack: 3-h5bd9786_13              conda-forge
    afterimage:                           1.21-ha9998ff_1003         conda-forge
    atk-1.0:                              2.36.0-h3371d22_4          conda-forge
    binutils_impl_linux-64:               2.36.1-h193b22a_2          conda-forge
    binutils_linux-64:                    2.36-hf3e587d_4            conda-forge
    bzip2:                                1.0.8-h7f98852_4           conda-forge
    c-ares:                               1.18.1-h7f98852_0          conda-forge
    ca-certificates:                      2021.10.8-ha878542_0       conda-forge
    cairo:                                1.16.0-ha00ac49_1009       conda-forge
    cffi:                                 1.15.0-py310h0fdd8cc_0     conda-forge
    cfitsio:                              4.0.0-h9a35b8e_0           conda-forge
    davix:                                0.8.0-hc88465f_2           conda-forge
    expat:                                2.4.2-h9c3ff4c_0           conda-forge
    fftw:                                 3.3.10-nompi_h77c792f_102  conda-forge
    font-ttf-dejavu-sans-mono:            2.37-hab24e00_0            conda-forge
    font-ttf-inconsolata:                 3.000-h77eed37_0           conda-forge
    font-ttf-source-code-pro:             2.038-h77eed37_0           conda-forge
    font-ttf-ubuntu:                      0.83-hab24e00_0            conda-forge
    fontconfig:                           2.13.1-hba837de_1005       conda-forge
    fonts-conda-ecosystem:                1-0                        conda-forge
    fonts-conda-forge:                    1-0                        conda-forge
    freetype:                             2.10.4-h0708190_1          conda-forge
    fribidi:                              1.0.10-h36c2ea0_0          conda-forge
    ftgl:                                 2.4.0-hbcb1f35_0           conda-forge
    gcc_impl_linux-64:                    9.4.0-h03d3576_11          conda-forge
    gcc_linux-64:                         9.4.0-h391b98a_4           conda-forge
    gdk-pixbuf:                           2.42.6-h04a7f16_0          conda-forge
    gettext:                              0.19.8.1-h73d1719_1008     conda-forge
    giflib:                               5.2.1-h36c2ea0_2           conda-forge
    gl2ps:                                1.4.2-h0708190_0           conda-forge
    glew:                                 2.1.0-h9c3ff4c_2           conda-forge
    graphite2:                            1.3.13-h58526e2_1001       conda-forge
    graphviz:                             2.50.0-h85b4f2f_1          conda-forge
    gsl:                                  2.7-he838d99_0             conda-forge
    gsoap:                                2.8.117-h90a1d37_0         conda-forge
    gtk2:                                 2.24.33-h539f30e_1         conda-forge
    gts:                                  0.7.6-h64030ff_2           conda-forge
    gwollum:                              3.1.0-hf25a816_0           conda-forge
    gxx_impl_linux-64:                    9.4.0-h03d3576_11          conda-forge
    gxx_linux-64:                         9.4.0-h0316aca_4           conda-forge
    harfbuzz:                             3.2.0-hb4a5f5f_0           conda-forge
    hdf5:                                 1.12.1-nompi_h2750804_103  conda-forge
    icu:                                  69.1-h9c3ff4c_0            conda-forge
    jbig:                                 2.1-h7f98852_2003          conda-forge
    jpeg:                                 9d-h36c2ea0_0              conda-forge
    kernel-headers_linux-64:              3.10.0-h4a8ded7_13         conda-forge
    krb5:                                 1.19.2-hcc1bbae_3          conda-forge
    ld_impl_linux-64:                     2.36.1-hea4e1c9_2          conda-forge
    lerc:                                 3.0-h9c3ff4c_0             conda-forge
    libblas:                              3.9.0-12_linux64_openblas  conda-forge
    libcblas:                             3.9.0-12_linux64_openblas  conda-forge
    libcurl:                              7.81.0-h2574ce0_0          conda-forge
    libcxx:                               12.0.1-h4bd325d_1          conda-forge
    libcxxabi:                            12.0.1-ha770c72_1          conda-forge
    libdeflate:                           1.8-h7f98852_0             conda-forge
    libedit:                              3.1.20191231-he28a2e2_2    conda-forge
    libev:                                4.33-h516909a_1            conda-forge
    libffi:                               3.4.2-h7f98852_5           conda-forge
    libframel:                            8.41.3-h7f98852_1          conda-forge
    libgcc-devel_linux-64:                9.4.0-hd854feb_11          conda-forge
    libgcc-ng:                            11.2.0-h1d223b6_11         conda-forge
    libgd:                                2.3.3-h3cfcdeb_1           conda-forge
    libgfortran-ng:                       11.2.0-h69a702a_11         conda-forge
    libgfortran5:                         11.2.0-h5c6108e_11         conda-forge
    libglib:                              2.70.2-h174f98d_1          conda-forge
    libglu:                               9.0.0-he1b5a44_1001        conda-forge
    libgomp:                              11.2.0-h1d223b6_11         conda-forge
    libiconv:                             1.16-h516909a_0            conda-forge
    liblapack:                            3.9.0-12_linux64_openblas  conda-forge
    libllvm9:                             9.0.1-default_hc23dcda_7   conda-forge
    libnghttp2:                           1.43.0-h812cca2_1          conda-forge
    libnsl:                               2.0.0-h7f98852_0           conda-forge
    libopenblas:                          0.3.18-pthreads_h8fe5266_0 conda-forge
    libpng:                               1.6.37-h21135ba_2          conda-forge
    librsvg:                              2.52.5-hc3c00ef_1          conda-forge
    libsanitizer:                         9.4.0-h79bfe98_11          conda-forge
    libssh2:                              1.10.0-ha56f1ee_2          conda-forge
    libstdcxx-devel_linux-64:             9.4.0-hd854feb_11          conda-forge
    libstdcxx-ng:                         11.2.0-he4da1e4_11         conda-forge
    libtiff:                              4.3.0-h6f004c6_2           conda-forge
    libtool:                              2.4.6-h9c3ff4c_1008        conda-forge
    libuuid:                              2.32.1-h7f98852_1000       conda-forge
    libwebp:                              1.2.1-h3452ae3_0           conda-forge
    libwebp-base:                         1.2.1-h7f98852_0           conda-forge
    libxcb:                               1.13-h7f98852_1004         conda-forge
    libxml2:                              2.9.12-h885dcf4_1          conda-forge
    libzlib:                              1.2.11-h36c2ea0_1013       conda-forge
    lz4-c:                                1.9.3-h9c3ff4c_1           conda-forge
    ncurses:                              6.2-h58526e2_4             conda-forge
    nlohmann_json:                        3.10.3-h9c3ff4c_0          conda-forge
    numpy:                                1.22.0-py310h454958d_0     conda-forge
    openssl:                              1.1.1l-h7f98852_0          conda-forge
    pango:                                1.48.10-h54213e6_2         conda-forge
    pcre:                                 8.45-h9c3ff4c_0            conda-forge
    pip:                                  21.3.1-pyhd8ed1ab_0        conda-forge
    pixman:                               0.40.0-h36c2ea0_0          conda-forge
    pthread-stubs:                        0.4-h36c2ea0_1001          conda-forge
    pycparser:                            2.21-pyhd8ed1ab_0          conda-forge
    pythia8:                              8.8-py310h122e73d_0        conda-forge
    python:                               3.10.2-h62f1059_0_cpython  conda-forge
    python_abi:                           3.10-2_cp310               conda-forge
    readline:                             8.1-h46c0cb4_0             conda-forge
    root_base:                            6.24.6-py310h57d1b7e_1     conda-forge
    scitokens-cpp:                        0.6.3-h00015f2_2           conda-forge
    setuptools:                           60.5.0-py310hff52083_0     conda-forge
    sqlite:                               3.37.0-h9cd32fc_0          conda-forge
    sysroot_linux-64:                     2.17-h4a8ded7_13           conda-forge
    tbb:                                  2021.5.0-h4bd325d_0        conda-forge
    tk:                                   8.6.11-h27826a3_1          conda-forge
    tzdata:                               2021e-he74cb21_0           conda-forge
    vdt:                                  0.4.3-h9c3ff4c_0           conda-forge
    vector-classes:                       1.4.2-h4bd325d_1           conda-forge
    wheel:                                0.37.1-pyhd8ed1ab_0        conda-forge
    xorg-fixesproto:                      5.0-h7f98852_1002          conda-forge
    xorg-kbproto:                         1.0.7-h7f98852_1002        conda-forge
    xorg-libice:                          1.0.10-h7f98852_0          conda-forge
    xorg-libsm:                           1.2.3-hd9c2040_1000        conda-forge
    xorg-libx11:                          1.6.12-h36c2ea0_0          conda-forge
    xorg-libxau:                          1.0.9-h7f98852_0           conda-forge
    xorg-libxcursor:                      1.2.0-h516909a_0           conda-forge
    xorg-libxdmcp:                        1.1.3-h7f98852_0           conda-forge
    xorg-libxext:                         1.3.4-h516909a_0           conda-forge
    xorg-libxfixes:                       5.0.3-h516909a_1004        conda-forge
    xorg-libxft:                          2.3.4-h8aea5ac_0           conda-forge
    xorg-libxpm:                          3.5.13-h516909a_0          conda-forge
    xorg-libxrender:                      0.9.10-h516909a_1002       conda-forge
    xorg-libxt:                           1.1.5-h516909a_1003        conda-forge
    xorg-renderproto:                     0.11.1-h7f98852_1002       conda-forge
    xorg-xextproto:                       7.3.0-h7f98852_1002        conda-forge
    xorg-xproto:                          7.0.31-h7f98852_1007       conda-forge
    xrootd:                               5.4.0-py310h2cd62ec_0      conda-forge
    xxhash:                               0.8.0-h7f98852_3           conda-forge
    xz:                                   5.2.5-h516909a_1           conda-forge
    zlib:                                 1.2.11-h36c2ea0_1013       conda-forge
    zstd:                                 1.5.1-ha95c52a_0           conda-forge


```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : base
    active env location : /home/duncan/opt/mambaforge
            shell level : 1
       user config file : /home/duncan/.condarc
 populated config files : /home/duncan/opt/mambaforge/.condarc
                          /home/duncan/.condarc
          conda version : 4.11.0
    conda-build version : 3.21.4
         python version : 3.9.7.final.0
       virtual packages : __linux=5.10.60.1=0
                          __glibc=2.28=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/duncan/opt/mambaforge  (writable)
      conda av data dir : /home/duncan/opt/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/duncan/opt/mambaforge/pkgs
                          /home/duncan/.conda/pkgs
       envs directories : /home/duncan/opt/mambaforge/envs
                          /home/duncan/.conda/envs
               platform : linux-64
             user-agent : conda/4.11.0 requests/2.26.0 CPython/3.9.7 Linux/5.10.60.1-microsoft-standard-WSL2 debian/10 glibc/2.28
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```
</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -146,7 +146,7 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ pin_compatible('xxhash', max_pin='x.x.x') }}
-    - nlohmann_json <3.10.4
+    - {{ pin_compatible('nlohmann_json', max_pin='x.x.x') }}
     - cffi
     - graphviz
     - {{ pin_compatible('vector-classes', max_pin='x.x') }}

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,7 +1,7 @@
 {% set name = "root" %}
 {% set tag_name = "6-24-06" %}
 {% set version = ".".join(tag_name.split("-")|map("int")|map("string")) %}
-{% set build_number = 1 %}
+{% set build_number = 2 %}
 {% set clang_version = "9.0.1" %}
 {% set clang_patches_version = "root_62400" %}
 

